libarchive/archive_read_disk_posix.c

```diff
@@ -828,7 +832,15 @@ _archive_read_data_block(struct archive *_a, const void **buff,
 		}
 	} else
 		bytes = 0;
-	if (bytes == 0) {
+	/*
+	 * Return an EOF unless we've read a leading empty sparse region, which
+	 * is used to represent fully-sparse files.
+	 *
+	 * TODO: it is not technically necessary to check for entry_tota == 0,
+	 * but this simplifies some unit tests that expect to only read data
+	 * regions with length > 0. Consider fixing these tests (test_sparse_basic).
+	*/
+	if (bytes == 0 && !(empty_sparse_region && t->entry_total == 0)) {
```

Comment:
I removed the entry_total == 0 check here and test_sparse_basic still passes for me.  Is the comment out-of-date?